src/main/java/com/rtbhouse/kafka/workers/impl/task/WorkerThread.java

```diff
@@ -107,13 +107,10 @@ private RecordStatusObserver createRecordStatusObserver(WorkerRecord<K, V> pollR
     }
 
     @Override
-    public void finish() {
+    public void close() {
```

Comment:
If taskManager.notifyTaskManager was added because of deadlock during rebalance, it will not work any more because close() is not called at the end of kafka-workers lifetime now.

Reply:
This notifyTaskManager() below is because setting stopped=true changes WorkerThread#isNotRunning which changes TaskManager#allThreadsNotRunning used in TaskManager#stopThreads.